data/src/main/java/org/apache/iceberg/data/TableMigrationUtil.java

```diff
@@ -121,9 +156,13 @@ private TableMigrationUtil() {
           datafiles[index] = buildDataFile(fileStatus.get(index), partitionKey, spec, metrics, "orc");
         });
       } else {
-        throw new UnsupportedOperationException("Unknown partition format: " + format);
+        if (skipCorruptFiles) {
+          LOG.warn("Unknown partition format: {}", format);
```

Comment:
Aren't we fine with just throwing an error here because we suppress them?

Reply:
It is not within the `Task`,  suppression exception does not take effect here.

Reply:
In the warn can you just add 
`"Skipping unknown partition format: {} - {}", format, partitionUri)`
Or whatever makes sense so a user has the full context here?